analyze.module

```diff
@@ -25,4 +29,232 @@ function analyze_theme() {
       ],
     ],
   ];
-}
\ No newline at end of file
+}
+
+/**
+ * Implements hook_menu_local_actions_alter().
+ */
+function analyze_menu_local_actions_alter(&$local_actions): void {
+
+  /** @var \Drupal\analyze\AnalyzePluginManager $service */
+  $service = \Drupal::service('plugin.manager.analyze');
+
+  // Add local actions for all entities if we have plugins enabled.
+  if ($plugins = $service->getDefinitions()) {
+    $plugin_routes = [];
+
+    foreach ($plugins as $plugin) {
+      foreach (analyze_get_entity_types() as $id => $route) {
+        $analyze_route = 'entity.' . $id . '.analyze';
+        $plugin_route = 'analyze.' . $id . '.' . $plugin['id'];
+        $plugin_routes[] = $plugin_route;
+
+        $local_actions[$plugin_route] = [
+          'id' => $plugin_route,
+          'title' => $plugin['label'],
+          'route_name' => $plugin_route,
+          'route_parameters' => [],
+          'appears_on' => [$analyze_route],
+          'class' => 'Drupal\Core\Menu\LocalActionDefault',
+          'provider' => 'analyze',
+          'options' => [],
+          'weight' => 0,
+        ];
+      }
+    }
+
+    // Add the plugins as local tasks on the plug report pages as well.
+    foreach ($plugin_routes as $route) {
+      if (!empty($local_actions[$route]['appears_on'])) {
+        $local_actions[$route]['appears_on'] = array_merge($local_actions[$route]['appears_on'], $plugin_routes);
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_local_tasks_alter().
+ */
+function analyze_local_tasks_alter(&$local_tasks): void {
+
+  /** @var \Drupal\analyze\AnalyzePluginManager $service */
+  $service = \Drupal::service('plugin.manager.analyze');
+
+  // Add local actions for all entities if we have plugins enabled.
+  if ($service->getDefinitions()) {
+    foreach (analyze_get_entity_types() as $id => $route) {
+      $analyze_route = 'entity.' . $id . '.analyze';
+
+      $local_tasks[$analyze_route] = [
+        'id' => $analyze_route,
+        'title' => t('Analyze'),
+        'route_name' => $analyze_route,
+        'route_parameters' => [],
+        'base_route' => $route,
+        'class' => 'Drupal\Core\Menu\LocalTaskDefault',
+        'provider' => 'analyze',
+        'options' => [],
+        'weight' => 0,
+      ];
+    }
+  }
+}
+
+/**
+ * Implements hook_menu_local_tasks_alter().
+ */
+function analyze_menu_local_tasks_alter(&$local_tasks, $route_name, RefinableCacheableDependencyInterface &$cacheability): void {
```

Comment:
Do we need this and `analyze_local_tasks_alter`? IIRC this is the effectively the pre render alteration, where as `analyze_local_tasks_alter` would be discovery?

Reply:
Both needed: one dynamically adds the top level tab (View | Edit | Delete | Analyze), the other adds the local tasks (each plugin's full report links)

Reply:
I _think_ with the 2 level tasks, this won't be needed

Reply:
Sorry: this one was adding the View | Edit | delete tabs for the entity when we were on a plugin full report page: I wasn't getting them before and took the code from a d.org issue trying to do the same thing on the node.revision route.